### DISCUSSION TITLE
What should be done before deactivating the old mediawiki?

### DISCUSSION THREAD

**Discussion by dpolivaev** [freeplane/docs#19] [General]
The old mediawiki is not good enough, but it still has a landing home page with information for newcomers, and it has categories grouping pages so that people have a chance of finding relevant content. 

Now the content is migrated, and I would like to ask you how the home page should change and if we need any replacement for category pages before freeplane.org can be linked with the new content.

Also I am not sure if linking to the "old wiki" content from the top of every page is needed and which more descriptive name we can give this content. "Old wiki" seems to be a term only a few people knowing its history can understand.

Any ideas are appreciated.

**Comment by macmarrum** [comment]
My proposal
* Old wiki → Pre-2021 wiki
* New wiki → Post-2021 wiki

Regarding categories, a static page could be added listing the migrated pages grouped by category.

The landing page could, for now, remain the same, i.e. https://freeplane.github.io/wiki/migrated-from-mediawiki/#/Home

But first, I would propose that people perform some verification of the migrated pages to decide if the migration/conversion from MediaWiki to Markdown is good enough, and report here any major issues.

What I have in mind is that there might be some additional work needed before linking to GitHub instead of MediaWiki, e.g. identifying and extracting missing media files, especially mind maps, and linking them in the migrated pages.
Alternatively, each page migrated from MediaWiki could have a link to the original page, so it could be easily viewed, if needed.

**Reply by dpolivaev** [reply]
@macmarrum  Could we have a call to talk about the migration process? What times/days work better for you?

**Reply by macmarrum** [reply]
@dpolivaev sure thing. I'm on Discord: `macmarrum#3214`. Let's continue there.

**Reply by sifran-github** [reply]
> @macmarrum Could we have a call to talk about the migration process? What times/days work better for you?

i would like to join as well.
next week sometime? during day time in europe?

**Comment by macmarrum** [comment]
Here are the main points that Dimitry and Mac discussed last night

Dimitry proposed
* to focus only on several pages to be migrated from MediaWiki to GitHub
  * Command-line_options_and_configuration.md
  * Add-ons_(install).md
  * Conditional_node_styles.md
  * LaTeX_in_Freeplane.md
  * Relationship_to_FreeMind.md
  * Spell_checker.md
  * Translation_How-To.md
  * Contributing_to_Freeplane's_Development.md
  * Scripting.md
  * Scripting!_Script_filter.md
  * Scripts_collection.md
  * Scripting!_Jython.md
  * Scripting_environment_setup.md
  * Export_Using_XSLT.md
  * Add-ons_(Develop).md
  * Git_howto.md
  * How_to_build_Freeplane.md
  * IDE_setup.md
  * Plugin_development.md
  * Mind_map_gallery.md (low priority)
* for the pages to be integrated into the new wiki
* to rename some of the new-wiki pages
* to use an `Attic` where draft pages will live until they are ready to be moved to the main wiki
* to put a concrete hierarchy of pages in the sidebar (3 levels), with level 1 as:
  * Home
  * Getting started
  * How to get help
  * User documentation
  * Mind map examples
  * Contribute
  * Scripting
  * Coding
  * Attic
* to decide which home-page content should be used → _Mac and Dimitry settled for the repo README.md with features-rundown images by Rexel_
* to rename the project from `wiki` → _Mac and Dimitry agreed on `docs`_

Mac will work to implement the changes. Mind_map_gallery.md is the most tricky one and will require the most work to save the mindmap files and images to disk and structure them nicely in Markdown. **Anyone wishing to help is welcome.**

**Reply by dpolivaev** [reply]
Thanks for the session and for the summary. After reading your post I would like to come up with the following suggestion about the mind map examples: I suggest removing them from the documentation entirely. I would keep only the mind maps eventually needed as an essential part of the documentation.

In the past, the embedding of interactive Freeplane mind maps in web pages using flash or Java Applets was possible, but now both technologies are no longer supported by the browsers. It means that we can make available only images, videos, pdfs, or mind map files for download. Because GitHub discussion forums would allow everyone to share their mind maps in any possible format I think it is the easier and better way to go. Currently, in https://github.com/freeplane/freeplane/discussions we already have a category called "Show and tell", but I could also add a dedicated category called "Shared mind maps". I have just added it.

So I think that the migration of example mind maps is not a part of the media wiki migration but a separate activity and that they do not need to become a part of the freeplane documentation project.

**Reply by dpolivaev** [reply]
The new category looks like that:
![Selection_094](https://user-images.githubusercontent.com/1558052/175242913-9b4e433d-124d-4596-bf97-bc1cdc495803.png)

**Reply by dpolivaev** [reply]
Oh, I found that it already has got content https://github.com/freeplane/freeplane/discussions/254

**Reply by dpolivaev** [reply]
So we could just link the docs to https://github.com/freeplane/freeplane/discussions/categories/shared-mind-maps

**Reply by macmarrum** [reply]
Sure thing. I like the idea 👍

**Comment by sifran-github** [comment]
I did post an initial scripting documentation pull request, https://github.com/freeplane/docs/pull/20, but i understand and support the choice not to merge it yet, because the old wiki just has much more information.

Is it possible that we schedule a discord talk about the documentation?

I need some help regarding changes i want to make. i'm not sure about:
1. What scripting documentation do we currently have?
2. How to write a new wiki? (if it is using forks and pull requests: then i think that i understand this point).
3. How to mark what i have done? mark/remove from the old wiki.... so lets say that i want to rearrange the current scripting documentation into new categories, and it will take a few weeks of work, and i want to mark which pages in the old wiki i have already converted... - how to do so?

My confusion is mainly about the status of the old wiki, and how to build the process of converting/changing it.

**Reply by macmarrum** [reply]
@sifran-github 

This is what I can answer already now:

1. Please have a look at the preview of the target site (after [the changes](https://github.com/freeplane/docs/discussions/19#discussioncomment-3007727) were introduced) https://macmarrum.github.io/freeplane-docs/
2. Yes, using forks and pull requests is fine
3. The old wiki is basically gone. Dimitry proposed to keep only selected pages. Please see [the comment above](https://github.com/freeplane/docs/discussions/19#discussioncomment-3007727)

**Comment by sifran-github** [comment]
Hello @macmarrum 

Dimitry and I are thinking about coordination of the freeplane docs project. We think that use of the main repository https://github.com/freeplane/docs could be easier for synchronization. if you have made changes to your local repository, wouldn't it be better to push it to freeplane/docs (either to main branch or to some other branch)? then we can pull it, edit it and add on top of your changes?

**Comment by sifran-github** [comment]
@macmarrum After talking to dimitry, I want to talk to you about the documentation. lets try to meet in discord, and schedule it?

**Comment by macmarrum** [comment]
Sifran, Dimitry and Mac discussed the topic again.

Dimitry explained that he had meant to have all pages migrated from MediaWiki, not only the ones listed in [the previous post](https://github.com/freeplane/docs/discussions/19#discussioncomment-3007727), and for them to end up in the `Attic`; the selected pages to be the priority for merging into `docs`.

Therefore Mac will rename `migrated-from-mediawiki` to `attic` – it contains imperfect pages. Sifran will write instructions about how people can join in: to review the attic pages, correct them, enhance and integrate into `docs`.

Here's the initial structure, including the priority pages:
```text
* [Home](/)
* [Getting started](getting-started.md)
  * [Command-line options and configuration](Command-line_options_and_configuration.md)
  * [Extend with add-ons](Add-ons_(install).md)
* [How to get help](how-to-get-help.md)
* [User documentation](user-documentation.md)
  * [First steps](first-steps.md)
  * [Working with mindmap files](working-w-mindmap-files.md)
  * [Formatting maps and nodes](formatting-maps-and-nodes.md)
  * [Styles](styles.md)
  * [Templates](templates.md)
  * [CSS for HTML/Markdown](css-for-html-or-markdown.md)
  * [Conditional node styles](Conditional_node_styles.md)
  * [LaTeX in Freeplane](LaTeX_in_Freeplane.md)
  * [Relationship to FreeMind](Relationship_to_FreeMind.md)
  * [Spell checker](Spell_checker.md)
  * References to other known documentation/tutorial sites
* [Mind map examples](https://github.com/freeplane/freeplane/discussions/categories/shared-mind-maps)
* [Contribute](contribute.md)
  * [Translation How-To](Translation_How-To.md)
  * [Contributing to Freeplane's Development](Contributing_to_Freeplane's_Development.md)
* [Scripting](Scripting.md)
  * Formulas
    * [Formulas](Formulas.md)
    * [Formula filter](Scripting!_Script_filter.md)
  * Scripts
    * [Scripts collection](Scripts_collection.md)
    * [Scripting! Jython](Scripting!_Jython.md)
    * [Scripting environment setup](Scripting_environment_setup.md)
  * XSLT for export/import
    * [Export Using XSLT](Export_Using_XSLT.md)
  * Add-ons
    * [Add-ons (Develop)](Add-ons_(Develop).md)
* Coding
  * [Git howto](Git_howto.md)
  * [How to build Freeplane](How_to_build_Freeplane.md)
  * [IDE setup](IDE_setup.md)
  * [Plugin development](Plugin_development.md)
* Attic (draft documents) <!-- as a separate docsify? -->
  * [Old home](old-Home.md)
  * [Many unsorted ideas related to Freeplane](many-unsorted-ideas-related-to-freeplane.md)
  * [Understand the Freeplane interface](understanding-interface.md)
  * [Understanding nodes](understanding-nodes.md)
  * [Customizing Freeplane](customizing-freeplane.md)
  * [Tutorials](/tutorials/index.md
```

Mac explained about docsify sidebar and sub-level sidebars. Dimitry was for introducing this concept, to make sure the side bar is short.

**Reply by macmarrum** [reply]
The site is reorganized as agreed → #28

**Reply by dpolivaev** [reply]
Is it ready to replace the MediaWiki as freeplane.org home?

**Reply by macmarrum** [reply]
I vote it's ready 😄
What do you think?

**Comment by sifran-github** [comment]
I want to emphasis that `Attic` is NOT a snapshot of the old wiki, and should contain pages that are NOT YET in the main branch. **It is a work in progress**. When we take a page, improve it and put it in the new wiki, we should delete it from the attic.

So:
- in the future, If someone wants to see **WHAT PAGES WERE IN THE OLD WIKI BUT ARE STILL NOT REFLECTED IN THE NEW DOCUMENTATION**, he can look at the attic. **Hopefully over time it will be empty.**
- in the future, if someone wants to see **HOW THE OLD WIKI LOOKED LIKE**, he can watch the old wiki, or if it is turned off, he can watch old versions of github somehow (some way to view a snapshot of the old wiki).

that's an important distinction on the purpose of the attic.

**Reply by macmarrum** [reply]
> When we take a page, improve it and put it in the new wiki, we should delete it from the attic.

I would say the file should remain, but the content should be replaced with the information **Migrated to docs** and a link to the new page, if possible.

> If someone wants to see WHAT PAGES WERE IN THE OLD WIKI BUT ARE STILL NOT REFLECTED IN THE NEW DOCUMENTATION, he can look at the attic. Hopefully over time it will be empty.

The way I see it, the list of pages that have been migrated or are remaining should be maintained in a dedicated file → `migration-from-attic-to-docs.md`

**Reply by sifran-github** [reply]
> I would say the file should remain, but the content should be replaced with the information Migrated to docs and a link to the new page, if possible.

I think we should at least move the file into a `moved_to_new_docs `folder. Otherwise it will be really hard to quickly understand what is left to do. We want people to be able to come, see what can be done, and help with it. When we keep a full list of all the old stuff, it will be confusing.

**Reply by macmarrum** [reply]
If a file is moved, the links no longer work. I would like to keep the attic functional as much as possible: for links from other pages to still work, showing placeholder pages after they'll have been moved.

To have a quick overview, people can have a look at `migration-from-attic-to-docs.md`

**Reply by dpolivaev** [reply]
I don't understand why you want to keep old pages as proxies in the attic and document specifically which pages have been moved. If you want to keep track of all changes you could add the file CHANGELOG.md containing the list of all important changes. If you don't want it the history is automatically contained as git commits and the commit messages.

**Reply by macmarrum** [reply]
@dpolivaev  I have explained why. If you ask a question referring a specific part of my explanation, I will be able to clarify it. Now I only know that you don't understand. That's too little for me to go on.

**Reply by sifran-github** [reply]
OK.
the purpose is to have a quick way to view the status of which pages were not moved yet.
I'm ok with a different technical solution for it, and not by using the file structure itself.

**Reply by dpolivaev** [reply]
> If a file is moved, the links no longer work. I would like to keep the attic functional as much as possible: for links from other pages to still work, showing placeholder pages after they'll have been moved.

If you edit the doc MD files locally in VS code you can easily find and replace all links when the file is moved or renamed. The proxy files in the attic are basically semantic noize.

**Reply by sifran-github** [reply]
I will just say that we need:

1. A way to see the content/pages that DID NOT MOVE yet to the new docs.
2. A way to see the content/pages that DID MOVE to the new docs.
3. Some way to see a snapshot of the old wiki, because why not (even if its as simple as a link to a zip file with all the pages).

I'm not tied to a specific technical solution, but it needs to include all points.

**Reply by dpolivaev** [reply]
> * A way to see the content/pages that DID NOT MOVE yet to the new docs.
> * A way to see the content/pages that DID MOVE to the new docs.

I would use separate folders for the attic and for each first level chapters like "user documentation". Then moving the file would be easily identifiable.

> * Some way to see a snapshot of the old wiki, because why not (even if its as simple as a link to a zip file with all the pages).

For having a preview of the old wiki snapshot you can either have another github project to see the rendered pages or run it locally on your machine. Macmarrum knows how.

**Reply by macmarrum** [reply]
> If you edit the doc MD files locally in VS code you can easily find and replace all links when the file is moved or renamed. The proxy files in the attic are basically semantic noize.

Given the proposed set-up — `attic` being a separate docsify site — it might be less straight-forward than you expect. But might be feasible. It's fine with me to try the move-and-re-link approach.

**Reply by dpolivaev** [reply]
> Given the proposed set-up — `attic` being a separate docsify site

Given context-dependent sidebars we could probably also include the attic as a usual first-level chapter.

**Reply by dpolivaev** [reply]
I mean I don' know what is better and think both is reasonable

**Reply by macmarrum** [reply]
I want search results to be specific to the area: either `docs` or `attic`, but not both at the same time.
`attic` is its own docsify site to keep it segregated from `docs`

**Reply by ofossum** [reply]
Why don't we use the project-boards of GitHub to track which site has been transferred from the attic to the new wiki?
It would be a Kanban (which can also be automated) to keep visible cards, every contributor is able to see and we can move around, depending on the status of a given page (or which moves automated).

See this introduction to this feature of GitHub for example: https://www.youtube.com/watch?v=idZyqNIrt84

**Comment by dpolivaev** [comment]
The new documentation site is available at https://docs.freeplane.org/

**Reply by macmarrum** [reply]
Cool 👍 

I can see that https://freeplane.github.io/docs/ redirects to https://docs.freeplane.org/

It's worth to note that `Attic` is broken. The implemented approach depends on the `Docs` site being located at `/docs/` and the `Attic` site at `/docs/attic/`.

I will work on a fix.

**Reply by macmarrum** [reply]
Path are now fixed (PR #29). The browser cache might need to be refreshed for the new content in pages to load correctly.

**Comment by dpolivaev** [comment]
I added https://docsify-darklight-theme.boopathikumar.me/#/installation directly to the main branch.
How do you like it?

**Reply by dpolivaev** [reply]
![Selection_096](https://user-images.githubusercontent.com/1558052/176024474-a5705125-e788-4439-85e0-818b342fc408.png)
![Selection_095](https://user-images.githubusercontent.com/1558052/176024477-97e1c2ea-0e20-4464-bb17-f91698703049.png)

**Reply by macmarrum** [reply]
It's interesting.

Personally, I use Dark Reader (browser extension) to make every page dark.

The issues that I see when using docsify-darklight-theme
* the theme defines fonts for body, code, etc but fails to import them, which results in the browser choosing the default (fallback) fonts: a serif font for body as can be seen in the screenshots; the same on my system, and monospace for code
* the site uses a different theme for `attic` to distinguish it from `docs`; the theme is light, and the transition form dark to light is quite shocking

Once the issues are fixed, I think the new theme will do nicely.

**Reply by dpolivaev** [reply]
If we all prefer dark themes I wish we also delivered them to our readers.
I have very little experience with the front-end and I am afraid I can not fix the issues you mentioned.

How could we solve it?

**Reply by macmarrum** [reply]
@dpolivaev In other words, for the time being, please change the theme back to the initial one.

Let's also try to keep to the agreement, that we use Pull Requests when doing changes, so that other contributors can opine.

And if there is a need to offer a public preview, let's do that in another repository (own fork or `freeplane/docs-development`, as discussed during the Friday discord call.

**Reply by dpolivaev** [reply]
All right, I reverted all related commits.

**Reply by dpolivaev** [reply]
I am trying to get the process right, so I forked the repo. Unfortunately, neither the link to attic works neither on your preview site https://macmarrum.github.io/freeplane-docs/#/ nor on my preview site https://dpolivaev.github.io/freeplane-docs/#/
Going to https://macmarrum.github.io/attic/ shows error 404.

Do you have an idea what happens and how it could be fixed?

**Reply by dpolivaev** [reply]
Finally, I discovered the missing pages under https://macmarrum.github.io/freeplane-docs/attic/#/ and https://dpolivaev.github.io/freeplane-docs/attic/#/

There is good news: in the version deployed on https://dpolivaev.github.io/freeplane-docs/attic/#/ switching the theme from light to dark has an effect also on  https://dpolivaev.github.io/freeplane-docs and vice versa

**Reply by dpolivaev** [reply]
I could fix the fonts and opened  a PR https://github.com/freeplane/docs/pull/32
Could you please check it at https://dpolivaev.github.io/freeplane-docs/#/ and https://dpolivaev.github.io/freeplane-docs/attic/#/ and merge it?

I still do not understand why the link to the attic does not work on our forks

@macmarrum

**Reply by macmarrum** [reply]
@dpolivaev Thanks. I've had a look; I'll create a PR against your repo with a few changes.

The link to **Attic** in the sidebar is static and absolute. It correctly points to **Attic** when the site sits in the root `/`, but fails when it's located in a directory e.g. `/freeplane-docs/`.
When doing a preview, we'll need to keep in mind the the link won't work. It needs to be entered manually in the location bar, e.g. `/freeplane-docs/attic/`, like you did.

**Reply by macmarrum** [reply]
Here's the PR with my proposed changes: https://github.com/dpolivaev/freeplane-docs/pull/1

**Comment by dpolivaev** [comment]
Currently, https://freeplane.org still redirects to the old wiki page at https://www.freeplane.org/wiki/index.php/Home

Do you think it should redirect to docs.freeplane.org instead?
What should happen with the  https://www.freeplane.org/wiki/index.php/Home? Keep it, hide it or change its subdomain?

**Reply by macmarrum** [reply]
Some documentation still references https://www.freeplane.org/wiki/, e.g. in the old discussion posts, probably in other places like youtube videos, articles and such. I believe those links should still lead somewhere meaningful.

At the same time, people should start using the new site.

I think that https://freeplane.org/ and https://www.freeplane.org/wiki/index.php/Home should redirect to https://docs.freeplane.org/.

It would be great if other pages at https://www.freeplane.org/wiki/ redirected or linked to their corresponding new pages, e.g. https://www.freeplane.org/wiki/index.php/Debugging_scripts -> https://docs.freeplane.org/#/Debugging_scripts

Alternatively, there should be a banner saying something like "The wiki has moved to http://docs.freeplane.org/, where it's being reorganized and improved."

**Reply by dpolivaev** [reply]
> Alternatively, there should be a banner saying something like "The wiki has moved to http://docs.freeplane.org/, where it's being reorganized and improved."

Do you think you could create a new markdown page under docs.freeplane.org saying just that the content old wiki has been moved here? Then I would just add redirects to the old site hosted on the Sourceforge because I control the .htaccess file there. It would be the easiest consistent and safe solution.

**Reply by macmarrum** [reply]
Now that I think about it, the old pages should not redirect automatically, or do so only after a delay so that the redirect can be stopped. This is because it will be useful to view the original pages by people who migrate from`Attic` to `Docs`.

And yes, I can create a `wiki-redirects-here` page in `Docs`.

**Reply by dpolivaev** [reply]
> Now that I think about it, the old pages should not redirect automatically, or do so only after a delay so that the redirect can be stopped

Do you think the old pages in the attic or the old pages in the wiki?
I don't know how to make a conditional redirect. It sounds hard to implement.
If you want to have the old wiki available for some reasons I would at least change the URL so that people can find it at a different place but all old links are consistently redirected to the new site.

**Reply by dpolivaev** [reply]
What do you think about having a discord call for better understanding?

**Reply by macmarrum** [reply]
I meant old mediawiki pages.

> If you want to have the old wiki available for some reasons I would at least change the URL so that people can find it at a different place but all old links are consistently redirected to the new site.

Sounds good to me 👍 

OK for a discord call, though I think all is clear now.

**Reply by dpolivaev** [reply]
I am in favor of the discord call. I am available now.

**Reply by dpolivaev** [reply]
If you can not make it now please let me know when you can.

**Reply by macmarrum** [reply]
I'm back. Now is fine for me

**Comment by dpolivaev** [comment]
Currently, page docs.freeplane.org does not reference its GitHub repository. Why not?
I think that it is kind of standard for the docsify rendered pages which makes the source markdown transparent and contributions a bit easier.

What do you think about restoring the link to the repository?

**Reply by macmarrum** [reply]
A good catch 👍  Given the agreed practice (fork - PR - merge), the "cat triangle" will be useful 😻 

But only for `docs`, not `attic` – would you agree?

If no objections, I'll introduce this small change directly.

**Reply by dpolivaev** [reply]
Why not for the attic too?
I am even not sure whether attic should become searchable, but it is a different question

**Reply by macmarrum** [reply]
To me the objective of the "cat triangle" is for people to have an instant link to the corresponding repository so that they can contribute.

`Docs` is the target site for contributions. I would reject any PR to change a page in `Attic` except for moving it to `Docs` and relinking other pages. This kind of activity will be done by people knowledgeable about the process, i.e. those who already know where the `Attic` repository is located.

I would like to learn about your rationale for displaying the triangle in `Attic`.

**Reply by dpolivaev** [reply]
I think we have still a bit different understanding of the function of the attic.
For me, the content in the attic is not obsolete. It has just not been finished or ported yet.
I think it makes sense to have it discoverable and allow PRs to the attic for all work-in-progress pages.
I also think that the obsolete pages can be removed unless you want to keep them just for having valid targets for old links.

**Reply by macmarrum** [reply]
> I think it makes sense to have it discoverable and allow PRs to the attic for all work-in-progress pages.

I see now what you mean. Fine with me. I can move the `Attic` pages to `Docs`, which will make them searchable. I'll work on retaining the banner, which might need some hacking.

**Comment by dpolivaev** [comment]
Hi @macmarrum, it seems that for me removing the redirect pages and fixing the links becomes easier after you integrate the attic with the other docs because of the differences how the links are currently handled in our forks and at the main site. So please go ahead and let me known when you are done.

**Reply by macmarrum** [reply]
👌 OK

**Reply by macmarrum** [reply]
@dpolivaev Done → PR #33

**Reply by dpolivaev** [reply]
I merged it. But what do you think about renaming the attic subfolder "wiki" to anything else because this name seems to be misleading and what do you think about renaming the attic home to anything else?

**Reply by macmarrum** [reply]
A good idea 💡 

I'd be happy to see your proposal for the names.

**Reply by dpolivaev** [reply]
Old wiki home
Drafts from Quickfold
?

**Reply by macmarrum** [reply]
`Old wiki home` – fine with me

`Drafts from Quickfold` – let's find something better, e.g. `post-2021 wiki`. This is because there were more people involved in creating and refining those pages than only Quickfold, and `drafts` is unclear, as discussed the other day.

**Reply by dpolivaev** [reply]
You probably know it better. Do you have any idea?

**Reply by dpolivaev** [reply]
I looked at the content. I would call it `Work in progress`

**Reply by macmarrum** [reply]
This is what I mean
```
attic
 |_post2021-wiki
```

I understand that this is your proposal
```
attic
 |_work-in-progress
```

I believe `work-in-progress` is too broad. Let's use `post2021-wiki`.

**Reply by dpolivaev** [reply]
I find post2021-wiki quite confusing and pretty bad naming because of the following reasons:

1. It does not communicate the intent
2. 2021 is a random number which nobody would understand even this year and even less likely in the following years. It does not reflect any function.
3. We agreed not to call the new site a wiki because it is not a wiki

I like work-in-progress because it at least reflects the intent

**Reply by dpolivaev** [reply]
"old wiki" reflects the origin and it makes it a bit better unless we find a better name for it too.

**Reply by macmarrum** [reply]
If one set of pages is located in work-in-progress, what does it say about the remaining pages in Attic? Not work-in-progress, one would be tempted to guess. We need to find a better solution.

I am troubled when I read in one post "_not to call the new site a wiki because it is not a wiki_" and in another a proposal to name the directory "old wiki".

My proposal is to move all pages from `attic/wiki` to `attic`, prefixing them with "-", i.e.
```
-customizing-freeplane.md
-freeplane-api-groovy.md
-glossary.md
-introduction-aka-many-unsorted-ideas-related-to-freeplane.md
-understanding-interface.md
-understanding-nodes.md
```
This way no new name needs to be agreed on, and the page can still be considered a separate group.

**Reply by dpolivaev** [reply]
What makes the pages so special that we group them?

**Reply by macmarrum** [reply]
They are part of the "new wiki" initiative; they were created in Markdown, not in MediaWiki, hence they weren't migrated and converted. While working on their content, one would be lost trying to locate their counterparts on the old MediaWiki site.

**Reply by dpolivaev** [reply]
It worked :)
What do you think about calling the subfolders just as you wrote "new wiki initiative" and "old mediawiki site"?

**Reply by macmarrum** [reply]
Yes, let's do like this 👍 

Therefore, the structure will look like this:
```
attic
 |_new-wiki-initiative
   |_ ...
 |_old-mediawiki-site
   |_ ...
```
I can implement the changes.

**Reply by dpolivaev** [reply]
Thanks, go on

**Reply by macmarrum** [reply]
PR #34 
@dpolivaev please have a look (preview: https://macmarrum.github.io/freeplane-docs/) and merge PR or comment

**Reply by dpolivaev** [reply]
I created a PR fixing the navigation from my fork into your devel branch.
The decision to create a dedicated attic/Home.md is intentional.

If you are fine with it please merge everything to the main repo, otherwise comment.

**Reply by macmarrum** [reply]
Thanks. Good ideas in the first commit.

I added [some more changes](https://github.com/freeplane/docs/pull/35/commits/431e13577e230e7abfff3ab76e05cbd30dfcce4a) to make it consistent with Docs
- lowercase file names
- indented bullets for directory items

Please review PR #35

**Reply by dpolivaev** [reply]
I agree that in the solution my second commit was not consistent with the layout in the main section.
We actually have a choice between the solution from PR #35 and applying the logic from the second commit to the sidebars in the home.
The approach in my second commit is probably more generic because it works with any number of sections with many subsections. It also has an advantage that the upper-level sections always keep their positions.

Which solution do you actually like more and why?

**Reply by macmarrum** [reply]
I like the way Docs sidebar directory items unfold, showing subitems.
This is like in a filesystem tree.
The approach is quite flexible. It supports several levels, as can be seen in [scripting](https://docs.freeplane.org/#/scripting/Scripting)

**Reply by dpolivaev** [reply]
Thanks, merged

**Comment by sifran-github** [comment]
i get errors trying to fetch the latest version to my local repository.
when i try to get it... i get the error message "This branch has conflicts that must be resolved".
i think that i need to use the command line interface in order to solve it somehow, but i'm not used to github.

**Reply by macmarrum** [reply]
That's to be expected. The structure of docs has changed. You will need to incorporate upstream changes to your fork/local first.

I would recommend learning a bit more about Git and GitHub before you continue. Maybe the following articles could give you a good start:

* https://www.freecodecamp.org/news/git-and-github-for-beginners/
* https://reflectoring.io/github-fork-and-pull/

There's a ton of other free tutorials and paid courses on the web for you to choose from. 

Git is a standard nowadays. You will benefit form learning how to operate it.

Once you're comfortable with Git, if you still have trouble moving forward with freeplane/docs, let me know and we can dive into details of what you're trying to do and where you got stuck.

**Comment by dpolivaev** [comment]
I configured all redirects and a favicon.
Now, https://freeplane.org and  https://www.freeplane.org are redirected to https://docs.freeplane.org.
Old wiki URLs are redirected https://docs.freeplane.org/#/after-redirect containing further instructions.

**Reply by macmarrum** [reply]
Excellent! 🍾 🎊

**Comment by dpolivaev** [comment]
I would like to disable the table-of-content insertions on many pages in the new docs but I could not figure out how to do it.

**Reply by macmarrum** [reply]
TOC is created globally for each page by javascript in `index.html`

I can see the following options to change the current behavior:
1. Maintain a TOC whitlist or a TOC blacklist of page names (in index.html)
2. Change the logic to create TOC only if a page contains `<div id="toc"></div>` and add the `div` only to those pages that should have TOC
3. Change the way TOC is displayed for all pages: add a "Show/Hide" button next to "Contents" and save the state with cookies/local storage; once hidden on one page, will be hidden on all

Option 3 can also be combined with 1 and 2 as an additional functionality.

I can implement the changes, or feel free to dabble in the front-end world yourself 😄

**Reply by dpolivaev** [reply]
Thanks for your explanations

> TOC is created globally for each page by javascript in `index.html`

Oh, I have not thought it might be your own implementation and have not looked there

> I can see the following options to change the current behavior:
> I can implement the changes, or feel free to dabble in the front-end world yourself smile

1. I am not sure if per-page opt-in or opt-out is better. If on your opinion the most pages should have the TOC, opt-out could make more sense. 
2. Docsify solves a similar problem when it creates a sidebar automatically. Look at 
https://docsify.js.org/#/more-pages?id=table-of-contents and https://docsify.js.org/#/more-pages?id=ignoring-subheaders
What do you think about adding page-specific comments to configure the TOC? Does it make sense, is it doable?

**Reply by macmarrum** [reply]
> Docsify solves a similar problem when it creates a sidebar automatically.

Please turn it on locally starting with `subMaxLevel: 6,` and see for yourself.
It doesn't play nice with the page tree we have in the sidebar.
The mixing of pages and anchors is confusing, in my opinion. That's why I opted for my own implementation.

> I am not sure if per-page opt-in or opt-out is better. If on your opinion the most pages should have the TOC, opt-out could make more sense.

I am actually leaning towards option 3, i.e. on top of the current way of injecting TOC to each page.

**Reply by dpolivaev** [reply]
I would like to have an additional opt-out maybe with combination with option 3.
As I created the https://docs.freeplane.org/#/after-redirect page it started with a heading, but I had to remove the heading just because I could not get rid of the TOC.

**Reply by macmarrum** [reply]
Fine with me. Please go ahead.
Alternatively, let me know to go ahead with the implementation.

**Reply by dpolivaev** [reply]
> Alternatively, let me know to go ahead with the implementation.

Yes, sure, go ahead. Many thanks.

**Reply by macmarrum** [reply]
Implemented – PR #39
Preview: https://macmarrum.github.io/freeplane-docs/#/after-redirect
I took the liberty to rephrase after-redirect and rename the directory to align it with the new description.
Please have a look.

**Comment by 1-DimensionalBox** [comment]
What fantastic progress with the documentation since our last community conference (28 May 2022)! Thank you 🙏

**Comment by 1-DimensionalBox** [comment]
A more practical comment: as redirection and favicon were [fixed by dpolivaev](https://github.com/freeplane/docs/discussions/19#discussioncomment-3112276), I think we can close this topic. It would be clearer if discussions about [Table of Content (TOC) management](https://github.com/freeplane/docs/discussions/19#discussioncomment-3112278) and the attic concept would get their own GH `issue` or GH `discussion`. Or do you think this is not necessary?
